packages/docusaurus/src/client/RouteAnnouncer.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {Portal} from './exports/Portal';
+
+type Props = {
+  // Force an update on route transition
+  location: string;
+  children: React.ReactNode;
+};
+
+type State = {
+  routeAnnouncement: string;
+};
+
+class RouteAnnouncerWrapper extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      routeAnnouncement: '',
+    };
+  }
+
+  override componentDidUpdate(): void {
+    const {routeAnnouncement} = this.state;
+    const {location} = this.props;
+
+    requestAnimationFrame(() => {
+      const firstHeading = document.querySelectorAll(`#__docusaurus h1`)[0];
+      const pageName =
+        firstHeading?.textContent ??
+        document.title ??
+        `new page at ${location}`;
```

Comment:
No need to fallback. `document.title` always exists because we have it in core `SiteMetadataDefaults`.